### DISCUSSION TITLE
Enviroment Variables problem in Flutter Web

### DISCUSSION THREAD

**Discussion by luciferro01** [vercel/community#3879] [Help]
### Summary

I have declared Enviroment variables in the vercel dashboard now I want to use them in my flutter app but nothing is working. But, I am unable to get these Enviroment variables

I have tried  js.context['process']['env']['NAME']   html.window.localStorage['NAME'] Platform.environment['NAME] as well but nothing is working out

I have also tried dotenv package as well but since I am not pushing it to the github it is not working in vercel after deploying the app

### Steps to Reproduce

Just try to fetch environment variables in flutter web application deployed on vercel

**Comment by jonasermert** [comment]
```
dependencies:
  dotenv: ^5.0.2
```

Then you need an .env file: 

`NAME=your_variable_value`

```
import 'package:flutter/material.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';

void main() async {
  await dotenv.load();
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    // Access the environment variable
    String name = dotenv.env['NAME'] ?? 'Default Value';

    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Text('Name: $name'),
        ),
      ),
    );
  }
}
```
add this to your .gitignore: .env

**Comment by rmahmadkhan** [comment]
@luciferro01 did you find any solution for this?

I have used simple `String.fromEnvironment('SOME_KEY')` in my Flutter app. Locally, I provide that through `--dart-define-from-file=.env/dev.json` tag but how do I achieve same through vercel env?

**Comment by luciferro01** [comment]
Nope not yet I didn't find any solution for this
